api/hasura/actions/_handlers/insertInteractionEvent.ts

```diff
@@ -0,0 +1,59 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { z } from 'zod';
+
+import { getProfilesWithAddress } from '../../../../api-lib/findProfile';
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { getInput } from '../../../../api-lib/handlerHelpers';
+import { errorResponse } from '../../../../api-lib/HttpError';
+
+// for json validation
+const literalSchema = z.union([z.string(), z.number(), z.boolean(), z.null()]);
+type Literal = z.infer<typeof literalSchema>;
```

Comment:
@sshmm how do we validate the input to this? Can a user send any interaction event they want and fake any data?